datadog-crashtracker/src/collector/process_handle.rs

```diff
@@ -1,46 +1,39 @@
 // Copyright 2025-Present Datadog, Inc. https://www.datadoghq.com/
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::shared::constants::DD_CRASHTRACK_MINIMUM_REAP_TIME_MS;
-use ddcommon::unix_utils::{reap_child_non_blocking, wait_for_pollhup};
+use ddcommon::unix_utils::{reap_child_non_blocking, wait_for_pollhup, TimeoutManager};
 use nix::unistd::Pid;
 use std::os::unix::io::RawFd;
-use std::time::Instant;
 
 pub(crate) struct ProcessHandle {
     pub uds_fd: RawFd,
-    pub pid: i32,
-    pub oneshot: bool,
+    pub pid: Option<i32>,
 }
 
 impl ProcessHandle {
-    pub fn new(uds_fd: RawFd, pid: i32, oneshot: bool) -> Self {
-        Self {
-            uds_fd,
-            pid,
-            oneshot,
-        }
+    pub fn new(uds_fd: RawFd, pid: Option<i32>) -> Self {
+        Self { uds_fd, pid }
     }
 
-    pub fn finish(&self, start_time: Instant, timeout_ms: u32) {
-        let pollhup_allowed_ms = timeout_ms
-            .saturating_sub(start_time.elapsed().as_millis() as u32)
-            .min(i32::MAX as u32) as i32;
-        let _ = wait_for_pollhup(self.uds_fd, pollhup_allowed_ms);
+    pub fn finish(&self, timeout_manager: &TimeoutManager) {
+        let result = wait_for_pollhup(self.uds_fd, timeout_manager);
+        debug_assert_eq!(result, Ok(true), "wait_for_pollhup failed: {result:?}");
 
-        if self.oneshot {
+        if let Some(pid) = self.pid {
```

Comment:
is `oneshot` the same as `we have Some(pid)` ? 🤔 

Before, when spawned from config, `oneshot` was false, so we would not enter here. Now, we have pid and we will execute this code.
Was it a bug ?

Reply:
Yes, I believe this was a bug.